andela/ah-robotics
# #162949066 Configure Authors Haven to use PostgreSQL as DB Engine

### What does this PR do?

Configures the application to use Postgres as the db engine

### Description of tasks to be completed?

Configure Authors Haven to use **PostgreSQL** as DB Engine instead of **SQLite**

### How should this be manually tested?

- Install or make sure PostgresDB is on your computer
- Clone this repo git clone https://github.com/andela/ah-robotics.git
- Checkout to the Chore branch `git checkout ch-change-db-postgres-162949066`
- Create a virtual environment with the command
> `virtualenv -p python3 env`
- Create a .env file and add the following db configurations:

```
DB_NAME='database_name'
DB_USER='created_username'
DB_PASS='password_for_created_user'
DB_HOST='host_name_eg_localhost'
DB_PORT='port_number_eg_5432'
```
Source the environment variables 
> `source .env`
#### Install dependencies 
> `pip install -r requirements.txt`

#### Setting Up the Database

Start Postgres Database server 
> `pg_ctl -D /usr/local/var/postgres start`
To open database on terminal use `psql Postgres`
Create a database with name `author_haven`
Migrate changes with `python manage.py makemigrations` then `python manage.py migrate`

Relevant Pivotal Tracker Stories

[#162948832](https://www.pivotaltracker.com/story/show/162949066)

#### Screenshots
> Updated README.md
<img width="1093" alt="screenshot 2019-01-08 at 14 27 46" src="https://user-images.githubusercontent.com/6994982/50828976-04c96980-1354-11e9-9e4a-ab28c3f3f7f3.png">

> Database Configuration on Settings.py

<img width="490" alt="screenshot 2019-01-08 at 14 28 00" src="https://user-images.githubusercontent.com/6994982/50829014-288caf80-1354-11e9-9132-6c72a6ba9e98.png">

> Created .env_example file

<img width="240" alt="screenshot 2019-01-08 at 14 28 20" src="https://user-images.githubusercontent.com/6994982/50829036-39d5bc00-1354-11e9-8104-5179e4b7b3a5.png">
